From 4138f6c8ca32d92fc1e163fccd71ecc3529e9715 Mon Sep 17 00:00:00 2001
From: Tycho Andersen <tycho@tycho.pizza>
Date: Sat, 2 Oct 2021 08:03:10 -0600
Subject: [PATCH 2/4] don't die on xcb's new "length" element

As with other new elements in the last 5 years, we don't actually *do*
anything with this besides ignore it, but we do need to explicitly ignore
it to avoid failures.

Signed-off-by: Tycho Andersen <tycho@tycho.pizza>
---
 generator/Data/XCB/Python/Parse.hs | 5 ++++-
 xcffib.cabal                       | 6 +++---
 2 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/generator/Data/XCB/Python/Parse.hs b/generator/Data/XCB/Python/Parse.hs
index dbae9ef..5c8926a 100644
--- a/generator/Data/XCB/Python/Parse.hs
+++ b/generator/Data/XCB/Python/Parse.hs
@@ -294,6 +294,7 @@ structElemToPyUnpack _ _ _ (Pad i) = Left (Nothing, mkPad i)
 structElemToPyUnpack _ _ _ (Doc _ _ _) = Left (Nothing, "")
 -- XXX: What does fd mean? we should implement it correctly
 structElemToPyUnpack _ _ _ (Fd _) = Left (Nothing, "")
+structElemToPyUnpack _ _ _ (Length _ _) = Left (Nothing, "")
 
 -- The switch fields pick the way to expression to pack based on the expression
 structElemToPyUnpack _ _ _ (Switch name expr _ bitcases) =
@@ -352,9 +353,10 @@ structElemToPyPack :: String
                                                                      [(Expr (), [GenStructElem Type])]
                                                     )]
 structElemToPyPack _ _ _ (Pad i) = Left (Nothing, mkPad i)
--- TODO: implement doc and fd?
+-- TODO: implement these?
 structElemToPyPack _ _ _ (Doc _ _ _) = Left (Nothing, "")
 structElemToPyPack _ _ _ (Fd _) = Left (Nothing, "")
+structElemToPyPack _ _ _ (Length _ _) = Left (Nothing, "")
 structElemToPyPack _ _ accessor (Switch n expr _ bitcases) =
   let name = accessor n
       cmp = xExpressionToPyExpr id expr
@@ -646,6 +648,7 @@ mkSyntheticMethod membs = do
       getName (Switch n _ _ _) = Just n
       getName (Doc _ _ _) = Nothing
       getName (Fd n) = Just n
+      getName (Length _ _) = Nothing
 
       assign :: String -> Statement ()
       assign n = mkAssign (mkDot "self" n) $ mkName n
diff --git a/xcffib.cabal b/xcffib.cabal
index ba1f4b1..8d5bdd6 100644
--- a/xcffib.cabal
+++ b/xcffib.cabal
@@ -25,7 +25,7 @@ source-repository head
 
 library
   build-depends: base ==4.*,
-                 xcb-types >= 0.10.0,
+                 xcb-types >= 0.11.0,
                  language-python >= 0.5.6,
                  filepath,
                  filemanip,
@@ -48,7 +48,7 @@ executable xcffibgen
                  xcffib,
                  language-python >= 0.5.6,
                  split,
-                 xcb-types >= 0.10.0,
+                 xcb-types >= 0.11.0,
                  optparse-applicative >= 0.13,
                  filepath,
                  filemanip,
@@ -83,7 +83,7 @@ test-suite GeneratorTests.hs
   type: exitcode-stdio-1.0
   build-depends: base ==4.*,
                  xcffib,
-                 xcb-types >= 0.10.0,
+                 xcb-types >= 0.11.0,
                  language-python >= 0.5.6,
                  HUnit,
                  test-framework,

